addons/road-generator/connectors/terrain3d_road_connector.gd

```diff
@@ -0,0 +1,220 @@
+	if not is_configured():
+		return
+	configure_road_update_signal()
+	for _container in road_manager.get_containers():
+		_container = _container as RoadContainer
+		var segs:Array = _container.get_segments()
+		refresh_roadsegments(segs)
+
+
+func _schedule_refresh(segments: Array) -> void:
+	_mutex.lock()
+	for _seg in segments:
+		# Using a dictionary to accumulate updates to process
+		_pending_updates[_seg] = true
+	if not is_instance_valid(_timer):
+		_timer = get_tree().create_timer(refresh_timer)
+	else:
+		pass
+	if not _timer.timeout.is_connected(_refresh_scheduled_segments):
+		_timer.timeout.connect(_refresh_scheduled_segments)
+	_mutex.unlock()
+
+
+func _refresh_scheduled_segments() -> void:
+	if Input.is_mouse_button_pressed(MOUSE_BUTTON_LEFT):
+		_mutex.lock()
+		_timer = null
+		call_deferred("_schedule_refresh", [])
+		_mutex.unlock()
+		return
+	_mutex.lock()
+	var _segs := _pending_updates.keys()
+	_pending_updates.clear()
+	_timer = null
+	_mutex.unlock()
+	refresh_roadsegments(_segs)
+	
+
+func refresh_roadsegments(segments: Array) -> void:
+	if not is_configured():
+		push_warning("Terrain-Road configuration invalid")
+		return
+	if not terrain.data:
+		push_warning("No terrain data available (yet)")
+		return
+	for _seg in segments:
+		_seg = _seg as RoadSegment
+		print("Refreshing %s/%s" % [_seg.get_parent().name, _seg.name])
+		flatten_terrain_via_roadsegment(_seg)
+	terrain.data.update_maps(Terrain3DRegion.MapType.TYPE_HEIGHT)
+
+
+# TODO: Move this utility into the RoadSegment (with offset) or RoadPoint class (no offset)
+func get_road_width(point: RoadPoint) -> float:
+	return (point.gutter_profile.x*2
+		+ point.shoulder_width_l
+		+ point.shoulder_width_r
+		+ point.lane_width * point.lanes.size()
+	)
+
+
+func flatten_terrain_via_roadsegment(segment: RoadSegment) -> void:
+	if not is_instance_valid(segment):
+		return
+	if not is_instance_valid(segment.start_point) or not is_instance_valid(segment.end_point):
+		return
+
+	var mesh := segment.road_mesh.mesh
+	if mesh == null:
+		return
+
+	var curve: Curve3D = segment.curve
+	var start_width: float = get_road_width(segment.start_point)
+	var end_width: float = get_road_width(segment.end_point)
+	var vertex_spacing: float = terrain.vertex_spacing
+
+	# Get global bounding box from mesh, expanded by affected smoothing radius
+	var local_aabb: AABB = mesh.get_aabb()
+	var offsets := Vector3(edge_margin+edge_falloff, 0, edge_margin+edge_falloff)
+	var aabb: AABB = segment.road_mesh.global_transform * segment.road_mesh.get_aabb()
+	var aabb_min := aabb.position - offsets
+	var aabb_max := aabb.position + aabb.size + offsets
+
+	# Snap bounds to terrain grid
+	var min := Vector3(aabb_min.x, 0, aabb_min.z).snapped(Vector3(vertex_spacing, 0, vertex_spacing))
+	var max := Vector3(aabb_max.x, 0, aabb_max.z).snapped(Vector3(vertex_spacing, 0, vertex_spacing)) + Vector3(vertex_spacing, 0, vertex_spacing)
+
+	var world_to_local := segment.global_transform.affine_inverse()
```

Comment:
For my own knowledge, why use `affine_inverse()` instead of `inverse()`?

From [the docs](https://docs.godotengine.org/en/stable/classes/class_transform3d.html#class-transform3d-method-affine-inverse):

> Returns the inverted version of this transform. Unlike `inverse()`, this method works with almost any `basis`, including non-uniform ones, but is slower. See also `Basis.inverse()`.

Can the `basis` for the segment be non-uniform (i.e. have scale applied to it)?